### PR TITLE
arch/esp32: Introduce the ESP32_RTC option.

### PR DESCRIPTION
**Pull request by Ouss4**
## Summary
Build the RTC modile when ESP32_RTC is enabled instead of the DRIVER_RTC option, because some of the features might be needed without the upperhalf driver.
## Impact
ESP32
## Testing
ESP32 defconfigs

### CODE REVIEW

File: arch/xtensa/src/esp32/esp32_rtc.c [hunk 1]
```diff
@@ -157,13 +157,11 @@
...
 }
 
-#ifdef CONFIG_RTC_DRIVER
 /* The magic data for the struct esp32_rtc_backup_s that is in RTC slow
  * memory.
  */
 
-#  define MAGIC_RTC_SAVE (UINT64_C(0x11223344556677))
-#endif
+#define MAGIC_RTC_SAVE (UINT64_C(0x11223344556677))
```
Comment (pkarashchenko) [inline arch/xtensa/src/esp32/esp32_rtc.c:164]: minor
```suggestion
#define MAGIC_RTC_SAVE UINT64_C(0x11223344556677)
```

File: arch/xtensa/src/esp32/Kconfig [hunk 2]
```diff
@@ -1423,6 +1389,48 @@ config ESP32_FREERUN
...
+
+	config ESP32_RTC_CLK_SRC_EXT_XTAL
+		bool "External 32kHz crystal"
+		select ESP_SYSTEM_RTC_EXT_XTAL
+
+	config ESP32_RTC_CLK_SRC_EXT_OSC
+		bool "External 32kHz oscillator at 32K_XN pin"
+
+	config ESP32_RTC_CLK_SRC_INT_8MD256
+		bool "Internal 8.5MHz oscillator, divided by 256 (~33kHz)"
```
Comment (pkarashchenko) [inline arch/xtensa/src/esp32/Kconfig:1429]: Please remove leading TABs for this block

File: arch/xtensa/src/esp32/esp32_rtc.c (hunk 1 above)
Comment (Ouss4) [reply arch/xtensa/src/esp32/esp32_rtc.c]: Fixed, thanks.

File: arch/xtensa/src/esp32/Kconfig (hunk 2 above)
Comment (Ouss4) [reply arch/xtensa/src/esp32/Kconfig]: Fixed, thanks.

**Review by pkarashchenko — APPROVED** [review]
LGTM!

**Pull request merged**